node/src/accountant/scanners/mod.rs

```diff
@@ -859,15 +859,23 @@ impl Scanner<RetrieveTransactions, ReceivedPayments> for ReceivableScanner {
                 "No newly received payments were detected during the scanning process."
             );
 
-            match self
-                .persistent_configuration
-                .set_start_block(Some(msg.new_start_block))
-            {
-                Ok(()) => debug!(logger, "Start block updated to {}", msg.new_start_block),
-                Err(e) => panic!(
-                    "Attempt to set new start block to {} failed due to: {:?}",
-                    msg.new_start_block, e
-                ),
+            if let Some(new_start_block) = msg.new_start_block {
+                let current_start_block = match self.persistent_configuration.start_block() {
+                    Ok(Some(current_start_block)) => current_start_block,
+                    _ => 0u64,
+                };
+                if new_start_block > current_start_block {
+                    match self
+                        .persistent_configuration
+                        .set_start_block(msg.new_start_block)
+                    {
+                        Ok(()) => debug!(logger, "Start block updated to {}", &new_start_block),
+                        Err(e) => panic!(
+                            "Attempt to set new start block to {} failed due to: {:?}",
+                            &new_start_block, e
+                        ),
```

Comment:
Maybe I'm the one overthinking. I occurred to me there might be a minor race condition (not something at the level I would expect Rust to catch) and I just thought it would be bad if blockchain_bridge had just advanced the block number and the accountant was still processing a previous block of transactions and  blindly updated the `start_block` to something old. This is just to prevent inadvertent retreat.